### ISSUE TITLE
sql: once a sequence is created, schema changes don't work any more

### ISSUE THREAD

**Issue by andreimatei** [cockroachdb/cockroach#21834]
There's something about the fact that sequences are not tables that makes some DDL statements useless. Look how the error changes after the sequence is created. In what follows, the fact that the index exists or not does not matter.

```
root@:26257/t> create database t;
CREATE DATABASE

Time: 8.277785ms

root@:26257/t> drop index xxx;
pq: index "xxx" not found

root@:26257/t> create sequence seq;
CREATE SEQUENCE

Time: 9.20511ms

root@:26257/t> drop index xxx;
pq: "t.seq" is not a table
```

**Comment by vilterp** [comment]
Wow, this is a weird one. Investigating.

**Comment by vilterp** [comment]
This is also a thing for views:

```
root@:26257/derp> create database d;
CREATE DATABASE

Time: 12.219ms

root@:26257/derp> use d;
SET

Time: 1.105ms

root@:26257/d> create table t (id int);
CREATE TABLE

Time: 12.229ms

root@:26257/d> create view v as select id from t;
CREATE VIEW

Time: 88.571ms

root@:26257/d> drop index xxx;
pq: "d.v" is not a table
```

**Comment by andreimatei** [comment]
If views have the same problem, we should prolly cherry-pick the fix on 1.1 if that release is just as broken.
